regression-test/suites/external_table_p0/hive/ddl/test_hive_ddl_and_ctas.groovy

```diff
@@ -217,75 +216,372 @@ suite("test_hive_ddl_and_ctas", "p0,external,hive,external_docker,external_docke
              (11, 'value_for_pt1', 'value_for_pt2'),
              (22, 'value_for_pt11', 'value_for_pt22');
             """
+        }
 
+        def destroySrcDDLForCTAS = { String catalog_name ->
             sql """ switch `${catalog_name}` """
-            // 1. external to external un-partitioned table
-            sql """ CREATE TABLE hive_ctas1 ENGINE=hive AS SELECT col1 FROM unpart_ctas_src; 
-            """
+            sql """ DROP TABLE IF EXISTS `test_ctas`.part_ctas_src """
+            sql """ DROP TABLE IF EXISTS `test_ctas`.unpart_ctas_src """
+            sql """ drop database if exists `test_ctas` """;
+            sql """ DROP TABLE IF EXISTS internal.test_ctas_olap.part_ctas_olap_src """
+            sql """ DROP TABLE IF EXISTS internal.test_ctas_olap.unpart_ctas_olap_src """
+            sql """ switch internal """;
+            sql """ drop database if exists test_ctas_olap """;
+        }
 
-            sql """ INSERT INTO hive_ctas1 SELECT col1 FROM unpart_ctas_src WHERE col1 > 1;
-            """
+        def test_ctas_tbl = { String file_format, String catalog_name ->
+            generateSrcDDLForCTAS(file_format, catalog_name)
+            try {
+                sql """ switch `${catalog_name}` """
+                // 1. external to external un-partitioned table
+                sql """ CREATE TABLE hive_ctas1 ENGINE=hive AS SELECT col1 FROM unpart_ctas_src; 
+                """
 
-            order_qt_ctas_01 """ SELECT * FROM hive_ctas1 """
-            sql """ DROP TABLE hive_ctas1 """
+                sql """ INSERT INTO hive_ctas1 SELECT col1 FROM unpart_ctas_src WHERE col1 > 1;
+                """
+                sql """ INSERT OVERWRITE TABLE hive_ctas1 SELECT col1 FROM unpart_ctas_src WHERE col1 > 1;
+                """
 
-            // 2. external to external partitioned table
-            sql """ CREATE TABLE hive_ctas2 ENGINE=hive AS SELECT col1,pt1,pt2 FROM part_ctas_src WHERE col1>0;
+                order_qt_ctas_01 """ SELECT * FROM hive_ctas1 """
+                sql """ DROP TABLE hive_ctas1 """
+
+                // 2. external to external partitioned table
+                sql """ CREATE TABLE hive_ctas2 ENGINE=hive AS SELECT col1,pt1,pt2 FROM part_ctas_src WHERE col1>0;
```

Comment:
why `hive_ctas2` is a partitioned table?

Reply:
see test_ctas_ex.hive_ctas2